test/integration/password_reset_test.rb

```diff
@@ -1,19 +1,25 @@
 require 'test_helper'
+include ActiveJob::TestHelper
 
 class PasswordResetTest < SystemTest
   setup do
     @user = create(:user, handle: nil)
+    @old_queue_adapter = queue_adapter
+    ActiveJob::Base.queue_adapter = :test
```

Comment:
Maybe I'm wrong, but by some quick debugging I found this:

`ActiveJob::Base.queue_adapter` defaults to `:inline` in test now. That means every job is triggered inline and taking some time. So all triggered jobs in tests are triggered inline.

Here you're setting adapter to `:test` which doesn't trigger job inline by default, but you can change that by `perform_enqueued_jobs`.

So actually `:inline` and `:test` with `perform_enqueued_jobs` will result into same result here.

By using `:test` adapter by default, you can trigger jobs you only need by `perform_enqueued_jobs` or use another helpers even without job triggering (assert it's in queue only).